packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2304,6 +2276,69 @@ class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     );
   }
 
+  async startListeningForPartitionChanges(model: IrisGridModel): Promise<void> {
+    log.debug('startListeningForPartitionChanges');
+    if (!isPartitionedGridModel(model) || !model.isPartitionRequired) {
+      return;
+    }
+    const { partitionConfig } = this.state;
+    if (partitionConfig !== undefined) {
+      // User already has a partition selected, just use that
+      return;
+    }
+
+    this.setState({
+      isSelectingPartition: true,
+      partitionConfig: {
+        mode: 'partition',
+        partitions: [],
+      },
+    });
+
+    log.debug('startListeningForPartitionChanges before keyTable');
+    const keyTable = await this.pending.add(
+      model.partitionKeysTable(),
+      resolved => resolved.close()
+    );
+    log.debug(
+      'startListeningForPartitionChanges after keyTable',
+      keyTable,
+      keyTable.size
+    );
+    const { dh } = model;
+
+    this.setState({ keyTable });
+
+    keyTable.addEventListener(
+      dh.Table.EVENT_UPDATED,
+      this.handlePartitionTableUpdate
+    );
+
+    const sorts = keyTable.columns.map(column => column.sort().desc());
+    keyTable.applySort(sorts);
+    // Select the last partition by default
+    keyTable.setViewport(0, 0);
+  }
+
+  stopListeningForPartitionChanges(
+    model: IrisGridModel,
+    resetState = false
+  ): void {
+    const { keyTable } = this.state;
```

Comment:
There is a race condition here - in `startListeningForPartitionChanges`, we fetch the keyTable asynchronously, so in theory it's possible that `stopListening` is called before `startListening` completes/can set the key table.

May need to keep that promise around explicitly such that it can be cancelled/unset if stopListening is called.